pkg/machinehealthcheck/reconciler.go

```diff
@@ -0,0 +1,109 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/clusterapi"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller/clientutil"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+)
+
+// Reconciler allows to reconcile machine health checks.
+type Reconciler struct {
+	client client.Client
+}
+
+// New returns a new Reconciler.
+func New(client client.Client) *Reconciler {
+	return &Reconciler{
+		client: client,
+	}
+}
+
+// InstallMachineHealthChecks installs machine health checks for a given cluster.
+func (r *Reconciler) InstallMachineHealthChecks(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster) error {
+	log.Info("Installing machine health checks for workload", "cluster", cluster.Name)
+
+	if cluster.Spec.MachineHealthCheck != nil {
+		setMachineHealthCheckTimeoutDefaults(cluster)
+	} else {
+
+		// check if machine health checks have already been installed on the cluster and populate the EKSA spec
+		mhc, err := r.machineHealthChecksExists(ctx, log, cluster)
+		if err != nil {
+			return err
+		}
+		if mhc != nil {
+			cluster.Spec.MachineHealthCheck = &anywherev1.MachineHealthCheck{
+				NodeStartupTimeout:      mhc.Spec.NodeStartupTimeout.OpenAPISchemaFormat(),
+				UnhealthyMachineTimeout: mhc.Spec.UnhealthyConditions[0].Timeout.String(),
```

Comment:
Do we need to have a null check for the array here?

Reply:
we don't really need a null check since the machine health check always populates 2 conditions, but i don't mind adding that in if you have a preference.

Reply:
I just want to make sure we don't run into exceptions, if machine health check dynamically populates then we should have the check, but it does make an assumption that the creator does that which can mess things up later when the assumption changes.